sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DeploymentStep.java

```diff
@@ -181,6 +181,13 @@ public synchronized void update(Protos.TaskStatus status) {
       case TASK_KILLED:
       case TASK_KILLING:
       case TASK_LOST:
+      //Tasks statuses relating to PARTITION_AWARE capability.
+      //Currently these Task statuses are handled in the same
+      //way as TASK_LOST.
+      case TASK_GONE:
+      case TASK_DROPPED:
+      case TASK_UNREACHABLE:
+      case TASK_GONE_BY_OPERATOR:
```

Comment:
Out of scope for this PR, but TASK_GONE_BY_OPERATOR, and other future states that automate pod replace should probably have some new status outside of pending to distinguish when a task was explicitly drained.